utils.go

```diff
@@ -91,22 +92,25 @@ func getQuerySnippet(req *http.Request) string {
 	io.Copy(ioutil.Discard, crc)
 	data := crc.String()
 
-	if req.Header.Get("Content-Encoding") != "gzip" {
+	ctype := getCompressType(req)
+	if len(ctype) == 0 || ctype == "lz4" {
 		return data
 	}
-
 	bs := bytes.NewBufferString(data)
-	gr, err := gzip.NewReader(bs)
-	if err != nil {
-		// It is better to return `gzipped` data instead
-		// of an empty string if the data cannot be ungzipped.
-		return data
-	}
+	// It is better to return compressed data instead
```

Comment:
This comment looks incorrect and misleading. There are two cases here:
1) The data cannot be decompressed at all. In this case it is better to return the compressed data instead of empty string. See lines `100-104` in the current code.
2) The data may be partially decompressed. In this case it is better returning the decompressed part of the data. See lines `106-109` in the current code.

So it would be better removing the comment, renaming `uncompress` into `tryDecompress` and moving the current logic from lines `99-109` into this function.
Then the code here would look like:
```go
b, err := tryDecompress(bs, ctype)
if err == nil {
    return string(b)
}

if len(b) > 0 {
    // It is better to return partially decompressed data instead of an empty string.
    return string(b)
}

// The data failed to be decompressed. Return compressed data
// instead of an empty string.
return data
```